src/TestSuite/Fixture/FixtureManager.php

```diff
@@ -227,14 +227,15 @@ protected function _loadFixtures($test)
     protected function _setupTable($fixture, $db, array $sources, $drop = true)
     {
         $configName = $db->configName();
-        if ($this->isFixtureSetup($configName, $fixture)) {
+        $isFixtureSetup = $this->isFixtureSetup($configName, $fixture);
+        if ($isFixtureSetup) {
             return;
         }
 
         $table = $fixture->sourceName();
         $exists = in_array($table, $sources);
 
-        if ($drop && $exists) {
+        if (($drop && $exists) || ($exists && !$isFixtureSetup)) {
```

Comment:
Should you also check $drop on the new arm? If the test has turned off dropTables we don't want to drop the tables even if the fixture isn't setup as its likely we don't have enough schema data to rebuild the table.